include/irr/asset/ICPUDescriptorSetLayout.h

```diff
@@ -31,8 +31,11 @@ class ICPUDescriptorSetLayout : public IDescriptorSetLayout<ICPUSampler>, public
                     if ((*cp->m_bindings)[i].samplers)
                         (*cp->m_bindings)[i].samplers = cp->m_samplers->begin() + ((*cp->m_bindings)[i].samplers - m_samplers->begin());
                 }
-                for (size_t i = 0ull; i < m_samplers->size(); ++i)
-                    (*cp->m_samplers)[i] = core::smart_refctd_ptr_static_cast<ICPUSampler>((*m_samplers)[i]->clone(_depth-1u));
+                if (cp->m_samplers)
```

Comment:
@Crisspl is this your fix?

Reply:
yea XD as i said, little bug in `clone()`